crates/ty_python_semantic/resources/mdtest/generics/pep695/functions.md

```diff
@@ -310,6 +310,106 @@ reveal_type(tuple_param("a", ("a", 1)))  # revealed: tuple[Literal["a"], Literal
 reveal_type(tuple_param(1, ("a", 1)))  # revealed: tuple[Literal["a"], Literal[1]]
 ```
 
+When a union parameter contains generic classes like `P[T] | Q[T]`, we can infer the typevar from
+the actual argument even for non-final classes.
+
+```py
+class P[T]:
+    x: T
```

Comment:
nit: I can't remember who I saw do this first, but I like adding a comment to indicate that this field is present to force the typevar to be invariant:


```suggestion
    x: T  # invariant
```